### PR TITLE
Fix nix flake by explicit version for `lsp-xxx` packages

### PR DESCRIPTION
**Pull request by guibou**
The nix flake was failing:

```
nix build .#haskell-language-server
```

Was failing because of bounds issue related to hackage `lsp`,
`lsp-types` and `lsp-test` projects.

I've bump theses version, introducing new flake input (so they can be
changed easily in the future) and now the build works.

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2557"></a>

### CODE REVIEW

**Review by jneira — COMMENTED** [review]
Thanks for fixing the nix build, will wait for a review from an actual nix user to merge

File: flake.nix
Comment (michaelpj) [inline flake.nix:39]: Cute. In the past we've just used an explicit fetchurl for this sort of thing, but this is nicer, I think. That said, it might be simpler to update if the flake input pointed to the corresponding tag in the github repository instead of hackage?

**Review by michaelpj — APPROVED** [review]

File: flake.nix
Comment (michaelpj) [reply flake.nix:39]: No obligation, I'll merge this in a day or so regardless.

Comment (guibou) [reply flake.nix:39]: Initially I tried to load it from github directly.. However I had problem finding the tags in the git repository. For example, there is no `1.4.0.0` tag (or any other variation, unless I'm wrong) in the https://github.com/haskell/lsp repository.

Then I realized that the source of truth is hackage. The `cabal` file for Haskell language server is actually using bounds based on hackage version and hence fetching the revision as it is on hackage seems more robust from my point of view.

Comment (michaelpj) [reply flake.nix:39]: The tags are confusing indeed, they're prefixed by the package. And it's no wonder you weren't able to find them... I forgot to push the tags (done now) :sweat_smile: anyway, not a big deal.

**Pull request merged**
